addons/web/static/tests/legacy/views/kanban_tests.js

```diff
@@ -542,6 +588,38 @@ QUnit.module('Views', {
         kanban.destroy();
     });
 
+    QUnit.test('prevent deletion when grouped by many2many field', async function (assert) {
```

Comment:
this one probably doesn't make sense. This mainly concerns x2manys, in which case the view isn't grouped. Here you mostly test the arch you wrote, and an implementation detail (`widget.deletable`).

Reply:
It is in use.
e.g. here https://github.com/odoo/odoo/blob/3fa3676240b48cce049d79f20292ac1eea8dd727/addons/crm/views/crm_lead_views.xml#L550